setup-tools/unlock-devices.sh

```diff
@@ -73,8 +73,18 @@ do
 
     echo "Start unlocking the device with ip address $DEVICE_IP"
 
+    # Check if the device is accessible
+    if ! ping -c 1 $DEVICE_IP &> /dev/null
+    then
+      echo "The IP $DEVICE_IP is not accessible. Cannot unlock the device with ip address: $DEVICE_IP"
+      exit
+    fi
+
```

Comment:
```suggestion
```
if the device is ping-able, but controller port 9091 is not reachable, this will fail

Reply:
Add the checking logic for the port 9091, please check it again. Thanks!

Reply:
I mean this if should be able to catch all these errors, and provide enough information in the error message, don't need an additional ip check

Reply:
Got it!